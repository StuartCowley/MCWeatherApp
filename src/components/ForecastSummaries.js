import React from "react";
import PropTypes from "prop-types";
import "../styles/App.css";
import ForecastSummary from "./ForecastSummary";

function ForecastSummaries({ forecasts }) {
  return (
    <div className="forecast-summaries">
      <h1>Forecast summaries</h1>
      {forecasts.map((forecast) => {
        return <ForecastSummary {...forecast} />;
      })}
    </div>
  );
}

ForecastSummaries.propTypes = {
  forecasts: PropTypes.arrayOf(PropTypes.object),
};

ForecastSummaries.defaultProps = {
  forecasts: [],
};

export default ForecastSummaries;
