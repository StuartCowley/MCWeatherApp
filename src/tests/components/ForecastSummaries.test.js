import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";

describe("ForecastDetails", () => {
  const validProps = [
    {
      date: 1525046400000,
      description: "Sunny",
      icon: "200",
      temperature: {
        max: 12,
        min: 10,
      },
    },
    {
      date: 1525046454300,
      description: "Cloudy",
      icon: "602",
      temperature: {
        max: 22,
        min: 15,
      },
    },
  ];

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastSummaries forecasts={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct number of ForecastSummary instances", () => {
    const { getAllByTestId } = render(
      <ForecastSummaries forecasts={validProps} />
    );
    expect(getAllByTestId("forecast-summary")).toHaveLength(2);
  });
});
