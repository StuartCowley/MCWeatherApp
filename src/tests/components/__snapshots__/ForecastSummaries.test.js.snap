// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Mon 30 Apr
      </div>
      <i
        class="wi wi-day-storm-showers wi-flip-horizontal wi-rotate-90 forecast-summary__icon"
        data-testid="forecast-icon"
        name="owm"
      />
      <div
        class="forecast-summary__max-temp"
      >
        12℃
      </div>
      <div
        class="forecast-summary__description"
      >
        Sunny
      </div>
      <button
        class="forecast-summary__details-button"
        type="button"
      >
        More details
      </button>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Mon 30 Apr
      </div>
      <i
        class="wi wi-day-snow wi-flip-horizontal wi-rotate-90 forecast-summary__icon"
        data-testid="forecast-icon"
        name="owm"
      />
      <div
        class="forecast-summary__max-temp"
      >
        22℃
      </div>
      <div
        class="forecast-summary__description"
      >
        Cloudy
      </div>
      <button
        class="forecast-summary__details-button"
        type="button"
      >
        More details
      </button>
    </div>
  </div>
</DocumentFragment>
`;
